components/PlacesAround.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from "react";
+import {
+  TouchableOpacity,
+  View,
+  Image,
+} from "react-native";
+import styles from '../constants/AppStyling';
+import Location from '../classes/location';
+import { REACT_APP_GOOGLE_PLACES_API } from "react-native-dotenv";
+
+type placeState = {
+  location:Location;
+ };
+
+type placeProps = {
+   lat:number;
+   long:number;
+   showPlaces(places:any[]):void;
+};
+class PlacesAround extends Component<placeProps,placeState> {
+  constructor(props) {
+    super(props);
+const {lat,long} = this.props;
+    this.state = {
+      location:new Location(lat,long),
+    };
+  }
+  async findPlaces(type:string){
+    const { location } = this.state;
+  const api = `https://maps.googleapis.com/maps/api/place/textsearch/json?query=${type}&radius=2000&location=${location.getLatitude()},${location.getLongitude()}&key=${REACT_APP_GOOGLE_PLACES_API}`;
+  const results = await fetch(api);
+  const json = await results.json();
+  const {showPlaces}=this.props;
+  showPlaces(json.results);
+
+}
+
+
+  render() {
+    return (
+      <View style={styles.optionsContainer}>
```

Comment:
For this, instead of hardcoding lots of different options, I would make an array of objects and store the image and title, then loop through that array to avoid unneeded code
ex: placesData = [{title: 'cafe', picture: require("../assets/cafe.png")}, {title: 'museum', picture: require("../assets/museum.png")}]

loop through this array and it will remove a lot of duplicate code

Reply:
resolved